internal/pkg/api/container/container.go

```diff
@@ -472,54 +472,42 @@ func ContainerRename(crp *wwapiv1.ContainerRenameParameter) (err error) {
 	return warewulfd.DaemonReload()
 }
 
-// Private helpers
+// create the system context and reading out environment variables
+func getSystemContext(noHttps bool, username string, password string) (sCtx *types.SystemContext, err error) {
+	sCtx = &types.SystemContext{}
+	// only check env if noHttps wasn't set
+	if !noHttps {
+		val, ok := os.LookupEnv("WAREWULF_OCI_NOHTTPS")
+		if ok {
 
-func setOCICredentials(sCtx *types.SystemContext) error {
-	username, userSet := os.LookupEnv("WAREWULF_OCI_USERNAME")
-	password, passSet := os.LookupEnv("WAREWULF_OCI_PASSWORD")
-	if userSet || passSet {
-		if userSet && passSet {
+			noHttps, err = strconv.ParseBool(val)
+			if err != nil {
+				return nil, fmt.Errorf("while parsing insecure http option: %v", err)
+			}
+
+		}
+		// only set this if we want to disable, otherwise leave as undefined
+		if noHttps {
+			sCtx.DockerInsecureSkipTLSVerify = types.NewOptionalBool(true)
+		}
+		sCtx.OCIInsecureSkipTLSVerify = noHttps
+	}
+	if username != "" {
+		username, _ = os.LookupEnv("WAREWULF_OCI_USERNAME")
```

Comment:
just to confirm, if the username is passed via CLI but the username is not set via `WAREWULF_OCI_USERNAME` (which means os.LookupEnv will return ''), the given username is still ignored here?

Reply:
Right, it must be  `passwd == ""` so that we check for the env value if no cli value was given.